src/entrystore/src/seg/resp.rs

```diff
@@ -39,15 +42,45 @@ impl Storage for Seg {
             ExpireTime::Seconds(n) => n,
             _ => 0,
         };
+        let value = match parse_signed_redis(set.value()) {
+            Some(integer) => U64(integer as u64),
+            None => Bytes(set.value()),
+        };
 
         if self
             .data
-            .insert(set.key(), set.value(), None, Duration::from_secs(ttl))
+            .insert(set.key(), value, None, Duration::from_secs(ttl))
             .is_ok()
         {
             Response::simple_string("OK")
         } else {
             Response::error("not stored")
         }
     }
+
+    fn incr(&mut self, incr: &Incr) -> Response {
+        if let Some(mut item) = self.data.get(incr.key()) {
```

Comment:
I'm not seeing any function being called which will modify the value in storage. I think with what you have currently, you'll keep getting the same result for incr?

Reply:
@brayniac 
![image](https://user-images.githubusercontent.com/833575/217623533-d9d84e1c-a011-4c94-91d8-649c420029d9.png)

`wrapping_add` on the underlying `Item` mutates the `Item` value https://github.com/pelikan-io/pelikan/blob/05e9086c179c855b54fd9218364b379f74376651/src/storage/seg/src/item/raw.rs#L187